src/terminal/input/terminalInput.cpp

```diff
@@ -513,209 +172,466 @@ TerminalInput::OutputType TerminalInput::HandleKey(const INPUT_RECORD& event)
+    defineKeyWithAltModifier(Ctrl + Shift + VK_BACK, ctrlBackSequence);
+
+    // TAB maps to HT, and Shift+TAB to CBT. The Ctrl modifier has no effect.
+    // The Alt modifier adds an ESC prefix, although in practice all the Alt
+    // mappings are likely to be system hotkeys.
+    const auto shiftTabSequence = fmt::format(FMT_COMPILE(L"{}Z"), _csi);
+    defineKeyWithAltModifier(VK_TAB, L"\t"s);
+    defineKeyWithAltModifier(Ctrl + VK_TAB, L"\t"s);
+    defineKeyWithAltModifier(Shift + VK_TAB, shiftTabSequence);
+    defineKeyWithAltModifier(Ctrl + Shift + VK_TAB, shiftTabSequence);
+
+    // RETURN maps to either CR or CR LF, depending on the Line Feed mode. With
+    // a Ctrl modifier it maps to LF, because that's the expected behavior for
+    // most PC keyboard layouts. The Alt modifier adds an ESC prefix.
+    const auto returnSequence = _inputMode.test(Mode::LineFeed) ? L"\r\n"s : L"\r"s;
+    defineKeyWithAltModifier(VK_RETURN, returnSequence);
+    defineKeyWithAltModifier(Shift + VK_RETURN, returnSequence);
+    defineKeyWithAltModifier(Ctrl + VK_RETURN, L"\n"s);
+    defineKeyWithAltModifier(Ctrl + Shift + VK_RETURN, L"\n"s);
+
+    // The keypad RETURN key works the same way, except when Keypad mode is
+    // enabled, but that's handled below with the other keypad keys.
+    defineKeyWithAltModifier(Enhanced + VK_RETURN, returnSequence);
+    defineKeyWithAltModifier(Shift + Enhanced + VK_RETURN, returnSequence);
+    defineKeyWithAltModifier(Ctrl + Enhanced + VK_RETURN, L"\n"s);
+    defineKeyWithAltModifier(Ctrl + Shift + Enhanced + VK_RETURN, L"\n"s);
+
+    // SPACE maps to SP, and Ctrl+SPACE to NUL. The Shift modifier as no effect.
+    // The Alt modifier adds an ESC prefix (not standard).
+    defineKeyWithAltModifier(VK_SPACE, L" "s);
+    defineKeyWithAltModifier(Shift + VK_SPACE, L" "s);
+    defineKeyWithAltModifier(Ctrl + VK_SPACE, L"\0"s);
+    defineKeyWithAltModifier(Ctrl + Shift + VK_SPACE, L"\0"s);
+
+    if (_inputMode.test(Mode::Ansi))
+    {
+        // F1 to F4 map to the VT keypad function keys, which are SS3 sequences.
+        // When combined with a modifier, we use CSI sequences with the modifier
+        // embedded as a parameter (not standard - a modern terminal extension).
+        defineKeypadKey(VK_F1, _ss3, L'P');
+        defineKeypadKey(VK_F2, _ss3, L'Q');
+        defineKeypadKey(VK_F3, _ss3, L'R');
+        defineKeypadKey(VK_F4, _ss3, L'S');
+
+        // F5 through F20 map to the top row VT function keys. They use standard
+        // DECFNK sequences with the modifier embedded as a parameter. The first
+        // five function keys on a VT terminal are typically local functions, so
+        // there's not much need to support mappings for them.
+        for (auto vk = VK_F5; vk <= VK_F20; vk++)
         {
-            return _makeCharOutput(0);
+            constexpr std::array parameters = { 15, 17, 18, 19, 20, 21, 23, 24, 25, 26, 28, 29, 31, 32, 33, 34 };

```

Comment:
With MSVC we need to make sure to always use `static constexpr` when inside functions: https://godbolt.org/z/bzs99endn

```suggestion
            static constexpr std::array<uint8_t, 16> parameters = { 15, 17, 18, 19, 20, 21, 23, 24, 25, 26, 28, 29, 31, 32, 33, 34 };
```